sound/soc/sof/intel/hda-dai.c

```diff
@@ -204,8 +205,6 @@ static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
 	else
 		link_bps = codec_dai->driver->capture.sig_bits;
 
-	snd_hdac_ext_stream_reset(hext_stream);
-
```

Comment:
@ujfalusi think about this: When you start stream for the first time ever, you dont need reset, but you need it after stopping. So previously we were doing a reset and start every time but now we do a setup during hw_params() for start and reset during cleanup() for stop. It does the same thing but a bit more logically

Reply:
@ranj063, all I'm saying that _all_ existing code does the `snd_hdac_ext_stream_reset()` first followed by `snd_hdac_ext_stream_setup()` and never reset the ext_stream _after_.

But the `snd_hdac_ext_stream_reset()` not only resets the HDA Stream Descriptor registers and FIFO but it will also make sure that the stream is out from reset before continuing.

If you say that it is granted that when we are here the SRTS bit for the stream is guarantied to be 0 (which it should be as 0 is the reset value) then OK, but I just wanted to note this change.